### ISSUE TITLE
Update (some) error messages 

### ISSUE THREAD

**Issue by AnaBelgun** [TerriaJS/terriajs#5459]
Currently when using a share link with non-working data layers, the error message is: 
![image](https://user-images.githubusercontent.com/18452161/116508673-31c72f00-a905-11eb-8caa-8c9ef661a1b5.png)

https://beta.nationalmap.terria.io/#share=s-qTv7YWTuhnOBDYGUuUedHEx6Ojh

The message needs a rethink to clarify the next steps for end users. 

Update 16 July 2021
Following a refinement chat on wording, we acknowledged that: 
1) understanding of the error handling architecture is not yet complete; and made more complicated because of v8 migration; hence decision was to re-start the implementation of sentry/rollbar for automatic tracking of errors followed by regular reviews to improve the error handling and messages; https://github.com/TerriaJS/terriajs/issues/5627 
2) in parallel with automatic tracking implementation, we're editing some of the most common error messages to be more suitable for (non-developers) end users. Rewording available in comments below.

**Comment by nf-s** [comment]
This PR partially fixes this - https://github.com/TerriaJS/terriajs/pull/5454

**Comment by wibily** [comment]
ticket is to get the PR into a state where it can be reviewed and merged in. Nick to update this ticket afterwards with what's misisng

**Comment by AnaBelgun** [comment]
Following action from update 16 July 2021. 

Groups of errors and draft messages. 

1) Network requests based errors, possibly caused by: 
- service down
- invalid response
- invalid configuration 
- other

Message: 
_A network request error occurred while accessing this application and data may not be available or displayed correctly due to data custodian's service outage. Please try reloading the map, accessing the data at a later stage, or selecting a different dataset. If problem persist, please submit feedback here. [link to feedback button]._ 

2) Map interaction
- tile error messages; data loads incorrectly

Message: 
_A map interaction error occurred while accessing this application and data may not be displaying correctly. Please try reloading the map, accessing the data at a later stage, or selecting a different dataset. If problem persists, please submit feedback here. [link to feedback button]._ 

3) Map initialisation (no error messages atm)
- loading issues 
- configuration can't be found
- Bing or Cesium missing/expired key causing globe not be available (black + stars screen with no globe)
- https://github.com/TerriaJS/terriajs/issues/5635

Message:
_A map initialisation error occurred while accessing this application. Please try reloading the map, change the base map or the 3D view to 2D from map settings. If problem persists, please submit feedback here [link to feedback button]._ 

4) Internet explorer support 
- Error message in relation to browser support - should this be included in 1) ?

**Comment by philipgrimmett** [comment]
Looks great @AnaBelgun. Suggested tweaks for brevity:

**A network request error occurred.**
Data may not be available or display correctly due to data custodian's service outage. Please try reloading the map, accessing the data at a later stage, or selecting a different dataset. If the problem persists, please submit feedback here. [link to feedback button].

**A map tiling error occurred.**
Data may not be displaying correctly. Please try reloading the map, accessing the data at a later stage, or selecting a different dataset. If the problem persists, please submit feedback here. [link to feedback button].

**A map initialisation error occurred.**
Please try reloading the map, [selecting an alternative base map](url), or [switching to 2D mode](url). If the problem persists, please submit feedback here [link to feedback button].

**Comment by t83714** [comment]
@philipgrimmett 
Considering people might use terriajs in a different way (e.g. customise UI, embedded in iframe with limited size etc. ), can we make the error message more generic (e.g. not reference any UI e.g. `[link to feedback button]`).
Is that possible to remove the line:
> If the problem persists, please submit feedback here [link to feedback button]

One example is this dataset:
https://data.nsw.gov.au/search/dataset/ds-nsw-ckan-29a03215-6e5b-4fb9-9efb-047c132c1a97/details
I actually need to capture the error message triggered by terriajs and display outside the embedding iframe.
People would be confused if we mention `feedback button`.

**Comment by philipgrimmett** [comment]
@t83714 good point, thanks! Error messages probably shouldn't print `If the problem persists, please submit feedback here [link to feedback button]` unless the map owner has enabled feedback or specified a support email address in their config.

**Comment by nf-s** [comment]
I have moved all Catalog related errors here:

![image](https://user-images.githubusercontent.com/6187649/126465093-aa3125a9-2a2e-407d-8c08-05f92386a7bc.png)

What do people think?

**Comment by t83714** [comment]
> I have moved all Catalog related errors here:
> 
> ![image](https://user-images.githubusercontent.com/6187649/126465093-aa3125a9-2a2e-407d-8c08-05f92386a7bc.png)
> 
> What do people think?

I think it's a nicer way of showing error message (better than popup box).
For this ticket, I am more concerns about the error message content / wording.
i.e. better make the error message wording more generic without any reference of terraijs specific UI (e.g. ask user to click any particular button etc.). Therefore, when integrated with terriajs or customise terraijs via terriamap, the error message can be captured, reused and displayed by other application in a different UI.

**Comment by AnaBelgun** [comment]
@t83714 the wording around user clicking on something, is to meet the requirement of giving the user some clarification on what to do next. In many cases, users are just faced with an error message without any help on what does that mean for them, and what they can do next (i.e. close this and try later or try different data, or write an email/feedback to send your error etc etc). 
Agree it has to be generic, but somehow we need to give a next step/action to the user as well. Do you have any ideas on how to balance the two?

**Comment by t83714** [comment]
@AnaBelgun  

I think "try again later" could be a good (and generic) next step instruction for users for most case already.

Regarding "ask user to report the error" as further step, I think we probably don't want to encourage users to report the error as:
- In geoSpatial visualisation area, the chance of server side / data source issue is higher  (probably my own opinion 😄 ). 
  - i.e. we can't do too much about it
  - even we want users to report the issues, it would be more appropriate to send to the data custodian.
  - give users the wrong impression that we are responsible for the servers hosting the datasets 
- The feedback button is visible on main UI. I think the user will eventually report the problem anyway even we didn't ask them to do so.

If we do want to provide the further instruction (in additional to the simple "try again later" instruction), do we think the following is an acceptable alternative option?

e.g. this error message:
```
Data may not be available or display correctly due to data custodian's service outage. Please try reloading the map, accessing the data at a later stage, or selecting a different dataset. If the problem persists, please submit feedback here. [link to feedback button].
```

We can change to:
```
Data may not be available or display correctly due to data custodian's service outage. Please try reloading the map, accessing the data at a later stage, or selecting a different dataset. If the problem persists, please report to the maintainer of the site.
```

**Comment by AnaBelgun** [comment]
thanks @t83714 not sure they would know who the maintainer of the site is. To avoid linking because of the issue you mentioned, shall we indicate where the feedback button is at least? something like:
"please report by using the Feedback button (lower right corner)."

**Comment by zoran995** [comment]
Our users would certainly be happy having a link to feedback button. When linking feedback button we should check if it's enabled and button is shown to user, this will resolve an issue for integrated map in nsw data catalog because it won't be available there. 
Also detect when terria is integrated via iframe show generic error message (without mentioning error message). 
p.s. We could also add some config parameters that will control this behavior (and urlParameter), and in the end there is language overrides to override the text of error message.

**Comment by t83714** [comment]
Thanks @AnaBelgun I think you are right. probably "the maintainer of the site"  could confuse some users.
Yeah if we do want to provide the instruction to report the error, I think the original error message (with link to feedback button) is OK.
We probably can't accommodate both integration use case and DT use case of terriajs with a single generic error message (which could not cover either cases well).
For integration use case, probably we can overwrite the error message using "language config" (as @zoran995 suggested) when build a customised terra-map site.

**Comment by AnaBelgun** [comment]
WOuld you mind raising a separate ticket just for the integration /language config so we don't loose sight please?

**Comment by nf-s** [comment]
@t83714 Yep totally agree. I haven't started refactoring error messages yet, just sorting out how Errors propagate through the codebase

**Comment by t83714** [comment]
@AnaBelgun 
I guess we probably already have a solution to override the language string without forking TerriaJS (@nf-s Please correct me if I were wrong) but I can't find the documentation on how to do it.
Therefore, I created a ticket about the documentation here: https://github.com/TerriaJS/terriajs/issues/5666

**Comment by nf-s** [comment]
Note to self:
 
- [x] Remove all errors with appName and email in random errors https://github.com/TerriaJS/terriajs/pull/5821
- [x] How to handle CORS error?  https://github.com/TerriaJS/terriajs/pull/5821

**Comment by wibily** [comment]
capped at estimate to prevent yak shaving

**Comment by nf-s** [comment]
## Completion of error message types

- [x] Network requests based errorshttps://github.com/TerriaJS/terriajs/pull/5821
- [ ] Map interaction
- [ ] Map initialisation (no error messages atm)

## Other possible error types

### Model configuration errors

We could use something like

#### Incorrect configuration error

This catalog model is incorrectly configured, it may not behave as expected. Please report this error by submitting feedback here
